pyparsing/core.py

```diff
@@ -3595,16 +3615,17 @@ def __init__(self):
         self.errmsg = "Expected end of line"
 
     def parseImpl(self, instring, loc, doActions=True):
-        if loc < len(instring):
```

Comment:
Please redo this section of changes in a separate PR. I think it actually changes the logic.